src/awkward/operations/ak_almost_equal.py

```diff
@@ -88,21 +97,24 @@ def visitor(left, right) -> bool:
             return False
 
         if left.is_list:
-            return (
-                backend.index_nplike.array_equal(left.starts, right.starts)
-                and backend.index_nplike.array_equal(left.stops, right.stops)
-                and visitor(
-                    left.content[: left.stops[-1]], right.content[: right.stops[-1]]
-                )
+            return backend.index_nplike.array_equal(
+                left.offsets, right.offsets
+            ) and visitor(
+                left.content[: left.offsets[-1]], right.content[: right.offsets[-1]]
```

Comment:
I didn't realise that all the list types now implement `offsets` too.

Reply:
`ListOffsetArray` and `RegularArray` are the only ones that have (guaranteed) contiguous `content`, so they're the only ones that can have an `offsets` property.

Note, however, that `RegularArray.offsets` has to create a buffer that didn't previously exist, so there's a performance cost.

Reply:
Indeed. I'm assuming that this function is not a performance-critical path (if we're using this function in testing). Given that we already use `ak.to_packed`, deviating from this assumption might warrant a more layout-aware implementation of this function.